servicetalk-http-router-jersey-internal/src/main/java/io/servicetalk/http/router/jersey/internal/BufferPublisherInputStream.java

```diff
@@ -86,7 +86,8 @@ public int read(final byte[] b, final int off, final int len) throws IOException
     public void offloadSourcePublisher(final HttpExecutionStrategy executionStrategy,
                                 final Executor executor) {
         if (inputStream == EMPTY_INPUT_STREAM) {
-            publisher = executionStrategy.offloadReceive(executor, publisher);
+            publisher = executionStrategy.isMetadataReceiveOffloaded() || executionStrategy.isDataReceiveOffloaded() ?
+                    publisher.publishOn(executor) : publisher;
```

Comment:
use conditional offloading here?

Reply:
Adding conditional offloading here causes about 18 test failures. I have left it out pending review of the tests.

Reply:
These tests failures all appear to be due to conditional offloading changes; double (or triple) offloading is eliminated. 

I will change the test assumptions so that the tests fail only if the executor is the IoExecutor.

```
Run JerseyRouterTestSuite with ServiceTalkJacksonSerializerAutoDiscoverable > JUnit Jupiter > ExecutionStrategyTest > testResource(TestExecutorStrategy, TestExecutorStrategy, TestExecutorStrategy, TestMode, String, RouterApi) > io.servicetalk.http.router.jersey.ExecutionStrategyTest.testResource(TestExecutorStrategy, TestExecutorStrategy, TestExecutorStrategy, TestMode, String, RouterApi)[257] FAILED
    java.lang.AssertionError: path=/rsc-rte-no-offloads/subrsc-rte-exec-id-single-mapped, router=EXEC, class=NO_OFFLOADS, method=EXEC, mode=POST_RS : {"rs-thread":"router-1730-2","exec":"ContextPreservingStExecutor{delegate=DefaultExecutor{executor=InternalExecutor{service=ExecutorService@4a4a4d4a}, scheduler=SingleThreadedScheduler{offload=Executor@4a4a4d4a}}}","thread":"route-1731-1"}
    Expected: a string starting with "route-"
         but: was "router-1730-2"
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at io.servicetalk.http.router.jersey.ExecutionStrategyTest.assertRouteExecutor(ExecutionStrategyTest.java:370)
        at io.servicetalk.http.router.jersey.ExecutionStrategyTest.runTest(ExecutionStrategyTest.java:300)
        at io.servicetalk.http.router.jersey.AbstractJerseyStreamingHttpServiceTest.runTwiceToEnsureEndpointCache(AbstractJerseyStreamingHttpServiceTest.java:369)
        at io.servicetalk.http.router.jersey.ExecutionStrategyTest.testResource(ExecutionStrategyTest.java:247)
```

```
JerseyRouterTestSuite > JUnit Jupiter > ExecutionStrategyTest > testResource(TestExecutorStrategy, TestExecutorStrategy, TestExecutorStrategy, TestMode, String, RouterApi) > io.servicetalk.http.router.jersey.ExecutionStrategyTest.testResource(TestExecutorStrategy, TestExecutorStrategy, TestExecutorStrategy, TestMode, String, RouterApi)[166] FAILED
    java.lang.AssertionError: path=/rsc-default/subrsc-rte-exec-id-publisher-mapped, router=DEFAULT, class=DEFAULT, method=EXEC, mode=POST_RS : {"rs-thread":"servicetalk-global-executor-3-3","exec":"ContextPreservingStExecutor{delegate=DefaultExecutor{executor=InternalExecutor{service=ExecutorService@a288c29}, scheduler=SingleThreadedScheduler{offload=Executor@a288c29}}}","thread":"route-1366-1"}
    Expected: a string starting with "route-"
         but: was "servicetalk-global-executor-3-3"
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at io.servicetalk.http.router.jersey.ExecutionStrategyTest.assertRouteExecutor(ExecutionStrategyTest.java:370)
        at io.servicetalk.http.router.jersey.ExecutionStrategyTest.runTest(ExecutionStrategyTest.java:275)
        at io.servicetalk.http.router.jersey.AbstractJerseyStreamingHttpServiceTest.runTwiceToEnsureEndpointCache(AbstractJerseyStreamingHttpServiceTest.java:369)
        at io.servicetalk.http.router.jersey.ExecutionStrategyTest.testResource(ExecutionStrategyTest.java:247)
```